src/openloco/windows/construction/OverheadTab.cpp

```diff
@@ -280,76 +244,34 @@ namespace openloco::ui::windows::construction::overhead
 
             if (gfx::clip_drawpixelinfo(&clipped, dpi, xPos, yPos, width, height))
             {
-                auto x = 0x2010;
-                auto y = 0x2010;
+                coord_t x = 0x2010;
+                coord_t y = 0x2010;
 
-                switch (gCurrentRotation)
-                {
-                    case 0:
-                    {
-                        auto bx = x;
-                        x = -x + y;
-                        y += bx;
-                        y >>= 1;
-                        y -= 460;
-                        break;
-                    }
-                    case 1:
-                    {
-                        x = -x;
-                        auto bx = x;
-                        x -= y;
-                        y += bx;
-                        y >>= 1;
-                        y -= 460;
-                        break;
-                    }
-                    case 2:
-                    {
-                        auto bx = x;
-                        x -= y;
-                        y = -y;
-                        y -= bx;
-                        y >>= 1;
-                        y -= 460;
-                        break;
-                    }
-                    case 3:
-                    {
-                        auto bx = x;
-                        x += y;
-                        y = -y;
-                        y += bx;
-                        y >>= 1;
-                        y -= 460;
-                        break;
-                    }
-                }
-                x -= (self->widgets[widx::image].width() / 2);
-                y -= ((self->widgets[widx::image].width() / 2) + 16);
-                clipped->x += x;
-                clipped->y += y;
+                auto rotCoord = rotate2DCoordinate({ x, y }, gCurrentRotation);
+                gfx::point_t screenPos = { rotCoord.y - rotCoord.x, ((rotCoord.x + rotCoord.y) >> 1) - 460 };
```

Comment:
You will need to insert some `static_cast<int16_t>(...)` around the two variables.